src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/service/StaffService.kt

```diff
@@ -15,24 +15,37 @@ class StaffService(
   private val staffRepository: StaffRepository,
 ) {
 
-  fun getOffenderAllocation(prisonNumber: String): Pair<StaffDetailResponse?, StaffDetailResponse?> {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun getOffenderAllocation(prisonNumber: String): Pair<StaffEntity?, StaffEntity?> {
     val offenderAllocation = allocationManagerService.getOffenderAllocation(prisonNumber)
 
     offenderAllocation?.let {
-      val primaryPom = nomisUserRolesService.getStaffDetail(it.primaryPom?.staffId.toString())
-      val secondaryPom = nomisUserRolesService.getStaffDetail(it.secondaryPom?.staffId.toString())
-
-      return Pair(primaryPom, secondaryPom)
+      return Pair(
+        fetchPomDetailsIfNotAlreadyExists(it.primaryPom?.staffId, prisonNumber, PomType.PRIMARY),
+        fetchPomDetailsIfNotAlreadyExists(it.secondaryPom?.staffId, prisonNumber, PomType.SECONDARY),
+      )
     } ?: throw BusinessException("No POM details found for $prisonNumber")
   }
 
-  fun saveStaffIfNotPresent(staffDetailResponse: StaffDetailResponse?) =
-    staffDetailResponse?.staffId?.let {
-      staffRepository.findByStaffId(it)
-        ?: return staffRepository.save(buildStaffEntity(staffDetailResponse))
+  fun fetchPomDetailsIfNotAlreadyExists(staffId: Int?, prisonNumber: String, pomType: PomType): StaffEntity? {
+    if (staffId == null) {
+      log.warn("No $pomType pom staffId found for $prisonNumber")
+      return null
+    }
+
+    val findByStaffId = staffRepository.findByStaffId(staffId)
```

Comment:
Naming : it's an odd name for a variable. Could you please rename.

Reply:
That's like the default suggestion for a variable in intellij. Updated and pushed